gamemode/cl_bannedmodels.lua

```diff
@@ -1,147 +1,186 @@
-GM.BannedModels = {}
+-- This file provides the functionality for accessing the banned models
+-- menu. It is used locally to determine if a model is viable (aka
+-- whether or not it gets an outline when moused over).
+
+
+GM.BannedModels = {} -- This is used as a hash table where the key is the model string and the value is true.
+local menu
+
 
 function GM:IsModelBanned(model)
-	return table.HasValue(self.BannedModels, model)
+	return self.BannedModels[model] == true
 end
 
+
 function GM:AddBannedModel(model)
-	table.insert(self.BannedModels, model)
+	if self.BannedModels[model] == true then return end -- Prevent duplicates.
```

Comment:
Not sure this is worth the code. It's not much more expensive to always write `true` or `nil` to the table.

Reply:
The issue here is that we don't want repeated calls to menu.AddModel because that will add duplicates to the menu. There needs to be a duplicate check somewhere and I figure it may as well be here.

Reply:
Oh, I didn't see the menu change below. Duh. I'd nix the comments about duplicates (and below about checking if it exists) as I was focused on that versus the actual result of not changing the menu around.